src/modules/users/users.controller.ts

```diff
@@ -319,9 +335,9 @@ export class UsersController {
     };
   }
 
-  @Post(':id/role')
+  @Patch(':id/role')
   @ApiBearerAuth()
-  @UseGuards(JwtAuthGuard, RolesGuard, UserIdGuard)
+  @UseGuards(JwtAuthGuard, UserIdGuard)
```

Comment:
Currently, an admin cannot change a user's role; only the user can do it for themselves. How will the admin update users' roles?

Reply:
Ok